k2/csrc/context.h

```diff
@@ -277,6 +277,35 @@ struct Region : public std::enable_shared_from_this<Region> {
     return reinterpret_cast<T *>(data);
   }
 
+  /* Extends the region (this is like realloc; and in fact, in future, we might
+     decide to use realloc-type things inside the implementation).
+        @param [in] new_bytes_used   New size of this region; if this is
+                         <= bytes_used nothing is done.  At exit, the
+                         bytes_used of this region will equal new_bytes_used.
+                         if num_bytes < new_bytes_used this region will be
+                         reallocated according to some heuristic (e.g. the
+                         larger of double the current size, or
+                         the next power of 2 greater than `new_bytes_used`). */
+  void Extend(size_t new_bytes_used) {
+    if (new_bytes_used <= bytes_used) return;
+    if (num_bytes < new_bytes_used) {  // reallocate and copy
+      size_t new_size = std::max<size_t>(num_bytes * 2, new_bytes_used);
+      size_t i = 4;
+      while (i < new_size / 8) i <<= 3;
```

Comment:
Not sure if you want to expose function 
https://github.com/k2-fsa/k2/blob/960abbcfa75655a858ae2c6e1f5daea0bb7b5470/k2/csrc/utils.cu#L66 
to be public as use it here.

Reply:
that has the wrong type... for now let's leave as-is.